lib/hdk_indexing_core/src/lib.rs

```diff
@@ -0,0 +1,20 @@
+use hdi::prelude::*;
+
+
+// Here, we use skip_no_mangle so that
+// this code is better suited for use as a library, than as a
+// zome directly itself as a crate. Elsewhere, in your zomes
+// you export the necessary extern, like so
+
+// Add the extern function that says how many links this zome has.
+// #[no_mangle]
+// pub fn __num_link_types() -> u8 {
+//     hdk_semantic_indexes_core::LinkTypes::len()
+// }
+
+#[hdk_link_types(skip_no_mangle = true)]
+pub enum LinkTypes {
+    IdentityAnchor,
+    SemanticIndex,
+    TimeIndex,
+}
```

Comment:
@pospi I'm moving your comment to the PR, as it's easier to track:

Pospi said
"hmm, I don't think there should be any situations where all 3 of these link types should be managed by the same controller zome. If this is just for Agent then it's possibly an architectural problem to resolve. I've been working steadily to separate those crates, ideally soon we can drop the semantic indexes dependency on hdk_records entirely and that will reduce this kind of confusion."

Reply:
My response...

But semantic and time indexing do get used in a colocated context right? So is it IdentityAnchor that would get isolated out?

Reply:
yep! thats right

Reply:
I'm going to leave it as a TODO I think

Reply:
addressed this now, it's gone from the lib